File: platform/packages_py/env_resolver/env_resolver/env_servicenow_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve


@dataclass(frozen=True)
class ServicenowEnv:
    base_url: str | None
    username: str | None
    password: str | None


def resolve_servicenow_env(config: dict[str, Any] | None = None) -> ServicenowEnv:
    return ServicenowEnv(
        base_url=resolve(None, ['SERVICENOW_BASE_URL'], config, 'base_url', None),
        username=resolve(None, ['SERVICENOW_USERNAME'], config, 'username', None),
        password=resolve(None, ['SERVICENOW_PASSWORD'], config, 'password', None),
    )

File: platform/packages_mjs/env-resolver/src/env-servicenow.x.ts
import { resolve } from '@internal/env-resolve';

export interface ServicenowEnv {
    baseUrl: string | undefined;
    username: string | undefined;
    password: string | undefined;
}

export function resolveServicenowEnv(config?: Record<string, any>): ServicenowEnv {
    return {
        baseUrl: resolve(undefined, ['SERVICENOW_BASE_URL'], config, 'baseUrl', undefined),
        username: resolve(undefined, ['SERVICENOW_USERNAME'], config, 'username', undefined),
        password: resolve(undefined, ['SERVICENOW_PASSWORD'], config, 'password', undefined),
    };
}
